tree/tree/src/TBasket.cxx

```diff
@@ -724,13 +724,56 @@ void TBasket::AdoptBuffer(TBuffer *user_buffer)
    fBufferRef = user_buffer;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Reset the read basket TBuffer memory allocation if needed.
+///
+/// This allows to reduce the number of memory allocation while avoiding to 
+/// always use the maximum size.
+
+void TBasket::ReadResetBuffer(Int_t basketnumber)
+{
+    // Downsize the buffer if needed.
+
+   const auto maxbaskets = fBranch->GetMaxBaskets();
+   if (!fBufferRef || basketnumber >= maxbaskets)
+      return;
+
+   Int_t curSize = fBufferRef->BufferSize();
+
+   const Float_t target_mem_ratio = fBranch->GetTree()->GetTargetMemoryRatio();
+   const auto basketbytes = fBranch->GetBasketBytes();
+
+   Int_t max_size = basketbytes[basketnumber];
+   for(Int_t b = basketnumber + 1; (b < maxbaskets) && (b < (basketnumber+10)); ++b) {
```

Comment:
The PR messages says that the look-ahead window is the next 3 baskets

Reply:
indeed. I tweak the code and not the text.